File: server/models/Conversations.js
const mongoose = require('mongoose');

const conversationSchema = mongoose.Schema({
    members: {
        type: Array,
        required: true
    },
    isGroup: {
        type: Boolean,
        default: false
    },
    groupId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Group',
        required: false
    },
    lastMessage: {
        message: String,
        sender: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User'
        },
        timestamp: {
            type: Date,
            default: Date.now
        },
        sequenceNumber: {
            type: Number,
            default: 0
        }
    }
}, {
    timestamps: true
});

const Conversation = mongoose.model('Conversation', conversationSchema)

module.exports = Conversation;

File: go_server/models/conversations.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type LastMessage struct {
	Message        string             `bson:"message,omitempty" json:"message"`
	Sender         primitive.ObjectID `bson:"sender,omitempty" json:"sender"`
	Timestamp      time.Time          `bson:"timestamp,omitempty" json:"timestamp"`
	SequenceNumber int64              `bson:"sequenceNumber,omitempty" json:"sequenceNumber"`
}

type Conversation struct {
	ID          primitive.ObjectID   `bson:"_id,omitempty" json:"id"`
	Members     []primitive.ObjectID `bson:"members" json:"members"`
	IsGroup     bool                 `bson:"isGroup,omitempty" json:"isGroup"`
	GroupID     *primitive.ObjectID  `bson:"groupId,omitempty" json:"groupId,omitempty"`
	LastMessage *LastMessage         `bson:"lastMessage,omitempty" json:"lastMessage,omitempty"`
	CreatedAt   time.Time            `bson:"createdAt,omitempty" json:"createdAt"`
	UpdatedAt   time.Time            `bson:"updatedAt,omitempty" json:"updatedAt"`
}